iXBRLViewerPlugin/__init__.py

```diff
@@ -150,4 +158,5 @@ def guiRun(cntlr, modelXbrl, attach, *args, **kwargs):
     'CntlrWinMain.Menu.Tools': toolsMenuExtender,
     'CntlrWinMain.Menu.View': viewMenuExtender,
     'CntlrWinMain.Xbrl.Loaded': guiRun,
+    'ModelDocument.IsPullLoadable': guiStartLogging,
```

Comment:
I'm not familiar with the Arelle plugin model.  It seems odd to do "real" work in an `Is` method; is this common practice, or is there a better even that can be used?  Does this need to stop logging at some point (i.e., does a long-running session keep accumulating messages in this handler)?  If you close/open repeatedly, does the plugin still work (i.e., new ModelXbrl and plugins are run against that)?

Reply:
Let me preface this with, I think you have raised some really good points, though there isn't a lot to go off of so I will just layout how I decided to do things and we can potentially come up with a better flow.

> It seems odd to do "real" work in an Is method; is this common practice, or is there a better even that can be used? 

I wouldn't say common, but of the one example I have to go off of someone using arelle similarly to the ixbrl-viewer (EdgarRenderer) I see this exact same thing being done. Herm added the same thing [1](https://github.com/Arelle/EdgarRenderer/blame/60d4b56fcd2a79356724196664d2e56b5e8b1251/__init__.py#L1565), [2](https://github.com/Arelle/EdgarRenderer/blame/60d4b56fcd2a79356724196664d2e56b5e8b1251/__init__.py#L1296-L1300) after making a change in arelle around the same time that [makes the buffer initialized to None](https://github.com/Arelle/Arelle/blame/fa0343705369ce6f0516ceafee993a53ee10eab0/arelle/CntlrWinMain.py#L1535-L1537).  Talking to Herm this morning, he confirmed that the EdgarRenderer was a good example to go off of, though also admitted the flow is a bit confusing so 🤷 . I will also point out that I think [failing because there is no log buffer is odd](https://github.com/Workiva/ixbrl-viewer/blob/7bcde691a5edd59487e9e3158554dffb14c7886c/iXBRLViewerPlugin/iXBRLViewer.py#L205) so perhaps just skipping and warning about the lack of validation messages not meaning your doc is necessarily clean would be better? The problem with that approach is that nothing in [arelle actually starts the buffer](https://github.com/Arelle/Arelle/blob/fa0343705369ce6f0516ceafee993a53ee10eab0/arelle/CntlrWinMain.py#L1535) so then there will never actually be validations and you will always get the warning

> Does this need to stop logging at some point (i.e., does a long-running session keep accumulating messages in this handler)? 

I haven't been able to make it crash yet but I suppose if you ran tons and tons of broken docs through it, it might crash so it'd probably be good to find an appropriate time to call stop, though I don't have a good example of when to call stop. The Edgar renderer seems to be doing it for [dubious reasons](https://github.com/Arelle/EdgarRenderer/blob/60d4b56fcd2a79356724196664d2e56b5e8b1251/__init__.py#L1444). I have the opposite fear that they have where I am worried about calling stop because everything shares that logger and calling stop will because the messages will randomly start going to something else in the middle of other operations potentially. Though maybe the same argument is true for calling `startLoggingBuffer`, though there was an error coded around this one.

> If you close/open repeatedly, does the plugin still work (i.e., new ModelXbrl and plugins are run against that)?

Yes

Reply:
Would `CntlrWinMain.Xbrl.Open` be suitable?

As far as I can see the log buffer does get appended to indefinitely.  I think it's probably necessary to clear the buffer when closing a document, as otherwise, validation messages relating to one document will be included against the next.  Also, if you run `Tools->Validation->Validate` then I suspect that this will cause validation messages to be doubled up.

I don't know how important it is to be able to included validation messages in GUI-generated viewers, but a simpler fix might be to pass `showValidations = False` to `createViewer` to disable this entirely in the GUI.  As far as I can tell there isn't a great way to get the validation messages associated with an individual report in Arelle; we're forced to grab the validation messages from the buffer from the appropriate period, which works OK from the command line when all we're doing is opening the document, but is a bit trickier in the GUI when the user may do other things that add messages to the buffer.

Reply:
> `showValidations = False` to `createViewer`

I'm a lot happier with that suggestion it surcumvents all of the complexity of the problem 👍 , I will make it so